src/generators.jl

```diff
@@ -632,16 +634,14 @@ function generalized_preferential_attachment_edges!(
             push!(edges, (v[1], i+1))
             i = i+1;
         elseif x < p+r #edge event, no self-loops permitted
-            loop = 1
-            while loop == 1
+            v1 = rand(edges)[1]
+            v2 = rand(edges)[1]
+            while (v1 == v2 && i != 1) #i != 1 because we want more than 1 node for this to work
```

Comment:
This change isn't correct, we should fix it.

Reply:
Take this example: `gpa_edges!(4,.75,.25,[(1,1)],1)` The old code may into an infinite loop here.
The idea in the new condition is, you can only add edges if you have more than one node => In this case this adds self loops to the one node to itself -- but my rationale was that this should be redundant if you already had just one node and one edge. Alternatively, we could throw an error if this function is started with less than two nodes. What do you think?

Reply:
Throw an error if there are less than two nodes and one edge.